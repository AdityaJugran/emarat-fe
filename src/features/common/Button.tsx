import React from 'react';

type PropsType = {
	type?: 'button' | 'submit';
	children: React.ReactNode;
	color?: 'primary' | 'danger' | 'warning' | 'success' | 'emarat-accent';
	loading?: boolean;
	onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
};

export default function Button({
	type = 'button',
	color = 'primary',
	loading = false,
	children,
	onClick,
}: PropsType) {
	return (
		<button
			type={type}
			className={`inline-block mt-6 py-3 px-10 bg-${color}-default hover:bg-${color}-hover active:bg-${color}-active text-white font-bold rounded-full`}
			onClick={onClick}
		>
			{children}
		</button>
	);
}
