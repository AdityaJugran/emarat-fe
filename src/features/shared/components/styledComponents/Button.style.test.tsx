import renderer from 'react-test-renderer';
import Button from 'features/shared/components/styledComponents/Button.style';

it('Styled Title Snapshot', () => {
	const tree = renderer.create(<Button />).toJSON();
	expect(tree).toMatchSnapshot();
});
