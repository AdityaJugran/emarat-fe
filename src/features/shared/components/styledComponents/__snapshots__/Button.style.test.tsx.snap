// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Title Snapshot 1`] = `
<button
  className="sc-bdfBQB ikXkus"
/>
`;
